blockindex/contractstaking/cache.go

```diff
@@ -299,28 +299,34 @@ func (s *contractStakingCache) getBucketTypeIndex(amount *big.Int, duration uint
 
 func (s *contractStakingCache) getBucketType(id uint64) (*BucketType, bool) {
 	bt, ok := s.bucketTypeMap[id]
-	return bt, ok
+	if !ok {
+		return nil, false
+	}
+	return bt.Clone(), ok
 }
 
 func (s *contractStakingCache) mustGetBucketType(id uint64) *BucketType {
 	bt, ok := s.getBucketType(id)
 	if !ok {
 		panic("bucket type not found")
 	}
-	return bt
+	return bt.Clone()
 }
 
 func (s *contractStakingCache) getBucketInfo(id uint64) (*bucketInfo, bool) {
 	bi, ok := s.bucketInfoMap[id]
-	return bi, ok
+	if !ok {
+		return nil, false
+	}
+	return bi.Clone(), ok
 }
 
 func (s *contractStakingCache) mustGetBucketInfo(id uint64) *bucketInfo {
 	bt, ok := s.getBucketInfo(id)
 	if !ok {
 		panic("bucket info not found")
 	}
-	return bt
+	return bt.Clone()
```

Comment:
no need, already cloned in `getBucketInfo`

Reply:
fixed